solr/core/src/test/org/apache/solr/search/TestFiltering.java

```diff
@@ -92,21 +93,34 @@ public void testLiveDocsSharing() throws Exception {
         QueryResult res = new QueryResult();
         searcher.search(res, cmd);
         set = res.getDocSet();
-        assertSame(set, live);
+        assertEffectivelySame(set.getFixedBitSet(), live.getFixedBitSet());
 
         cmd.setQuery(QParser.getParser(qstr + " OR id:0", null, req).getQuery());
         cmd.setFilterList(QParser.getParser(qstr + " OR id:1", null, req).getQuery());
         res = new QueryResult();
         searcher.search(res, cmd);
         set = res.getDocSet();
-        assertSame(set, live);
+        assertEffectivelySame(set.getFixedBitSet(), live.getFixedBitSet());
       }
 
     } finally {
       req.close();
     }
   }
 
+  /** If the a XOR b == 0, then both a & b are effectively the same bitset */
+  private void assertEffectivelySame(FixedBitSet a, FixedBitSet b) {
```

Comment:
Can you explain why assertSame is no longer a working in this test?  I remember working on this with Yonik.  The very point of this test is to test that we use the actual instance of live docs (notice the test method's name)

Reply:
It seems the change is no longer needed now, #2589 proposes to revert it.